File: app/src/main/java/com/ipd/taxiu/ui/activity/referral/ReferralCodeActivity.java
package com.ipd.taxiu.ui.activity.referral;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.ipd.taxiu.R;
import com.ipd.taxiu.ui.BaseUIActivity;
import com.ipd.taxiu.widget.ChooseFriendDialog;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Created by Miss on 2018/7/25
 * 我的推荐码
 */
public class ReferralCodeActivity extends BaseUIActivity implements View.OnClickListener{
    private Button btn_invite_friends;
    private TextView tv_friend_list,tv_recommend_earning;
    @Override
    protected int getContentLayout() {
        return R.layout.activity_referral_code;
    }

    @Override
    protected void initView(@Nullable Bundle bundle) {
        initToolbar();
        btn_invite_friends = findViewById(R.id.btn_invite_friends);
        tv_friend_list = findViewById(R.id.tv_friend_list);
        tv_recommend_earning = findViewById(R.id.tv_recommend_earning);
    }

    @Override
    protected void loadData() {

    }

    @Override
    protected void initListener() {
        btn_invite_friends.setOnClickListener(this);
        tv_friend_list.setOnClickListener(this);
        tv_recommend_earning.setOnClickListener(this);
    }

    @NotNull
    @Override
    protected String getToolbarTitle() {
        return "我的推荐码";
    }

    @Override
    public void onClick(View v) {
        Intent intent;
        switch (v.getId()){
            case R.id.btn_invite_friends:
                ChooseFriendDialog dialog = new ChooseFriendDialog(this,R.style.recharge_pay_dialog,1);
                dialog.show();
                break;
            case R.id.tv_friend_list:
                intent = new Intent(this,FriendListActivity.class);
                startActivity(intent);
                break;
            case R.id.tv_recommend_earning:
                intent = new Intent(this,RecommendEarningsActivity.class);
                startActivity(intent);
                break;
        }
    }
}

File: app/src/main/java/com/ipd/taxiu/ui/activity/referral/ReferralCodeActivity.kt
package com.ipd.taxiu.ui.activity.referral

import android.content.Intent
import android.os.Bundle
import cn.sharesdk.framework.Platform
import com.ipd.taxiu.R
import com.ipd.taxiu.bean.BaseResult
import com.ipd.taxiu.bean.RecommendInfoBean
import com.ipd.taxiu.imageload.ImageLoader
import com.ipd.taxiu.platform.global.Constant
import com.ipd.taxiu.platform.global.GlobalParam
import com.ipd.taxiu.platform.http.ApiManager
import com.ipd.taxiu.platform.http.HttpUrl
import com.ipd.taxiu.platform.http.Response
import com.ipd.taxiu.platform.http.RxScheduler
import com.ipd.taxiu.ui.BaseUIActivity
import com.ipd.taxiu.utils.ShareCallback
import com.ipd.taxiu.widget.ShareDialog
import com.ipd.taxiu.widget.ShareDialogClick
import kotlinx.android.synthetic.main.layout_referral_code.*
import java.util.HashMap

/**
 * Created by Miss on 2018/7/25
 * 我的推荐码
 */
class ReferralCodeActivity : BaseUIActivity() {
    override fun getToolbarTitle() = "我的推荐码"

    override fun getContentLayout() = R.layout.activity_referral_code

    override fun initView(bundle: Bundle?) {
        initToolbar()
    }

    override fun loadData() {
        showProgress()
        ApiManager.getService().recommendInfo(GlobalParam.getUserIdOrJump())
                .compose(RxScheduler.applyScheduler())
                .subscribe(object : Response<BaseResult<RecommendInfoBean>>() {
                    override fun _onNext(result: BaseResult<RecommendInfoBean>) {
                        if (result.code == 0) {
                            val info = result.data
                            ImageLoader.loadAvatar(mActivity, info.LOGO, civ_my_header)
                            header_name.text = info.NICKNAME
                            tv_recommend_code.text = "推荐码:${info.MY_CODE}"
                            ImageLoader.loadNoPlaceHolderImg(mActivity, info.TWO_CODE, iv_qr_code)


                            btn_invite_friends.setOnClickListener {
                                val dialog = ShareDialog(mActivity)
                                dialog.setShareDialogOnClickListener(ShareDialogClick()
                                        .setShareTitle(Constant.SHARE_INVITE_TITLE)
                                        .setShareContent(Constant.SHARE_INVITE_CONTENT)
                                        .setShareLogoUrl("")
                                        .setShareUrl(HttpUrl.APK_DOWNLOAD_URL)
                                        .setCallback(object : ShareDialogClick.MainPlatformActionListener {
                                            override fun onComplete(platform: Platform?, i: Int, hashMap: HashMap<String, Any>?) {
                                                toastShow(true, "分享成功")
                                                ShareCallback.shareUser()
                                            }

                                            override fun onError(platform: Platform?, i: Int, throwable: Throwable?) {
                                                toastShow("分享失败")
                                            }

                                            override fun onCancel(platform: Platform?, i: Int) {
                                                toastShow("取消分享")
                                            }

                                        }))
                                dialog.show()

                            }

                            showContent()
                        } else {
                            showError(result.msg)
                        }
                    }

                    override fun onError(e: Throwable?) {
                        showError()
                    }
                })


    }

    override fun initListener() {
        tv_friend_list.setOnClickListener {
            intent = Intent(this, FriendListActivity::class.java)
            startActivity(intent)
        }
        tv_recommend_earning!!.setOnClickListener {
            intent = Intent(this, RecommendEarningsActivity::class.java)
            startActivity(intent)
        }
    }


}
